src/NumericalAlgorithms/DiscontinuousGalerkin/FluxCommunicationTypes.hpp

```diff
@@ -39,18 +44,24 @@ struct FluxCommunicationTypes {
   using normal_dot_fluxes_tag =
       db::add_tag_prefix<Tags::NormalDotFlux, typename system::variables_tag>;
 
-  /// Variables tag for storing the magnitude of the face normal in
-  /// the LocalData.
-  struct MagnitudeOfFaceNormal {
-    using type = Scalar<DataVector>;
-  };
-
-  /// The type of the Variables needed for the local part of the flux
+  /// The type of the data needed for the local part of the flux
   /// numerical flux computations.  Contains the PackagedData, the
   /// normal fluxes, and MagnitudeOfFaceNormal.
-  using LocalData = Variables<tmpl::remove_duplicates<tmpl::append<
-      typename db::item_type<normal_dot_fluxes_tag>::tags_list,
-      typename PackagedData::tags_list, tmpl::list<MagnitudeOfFaceNormal>>>>;
+  struct LocalData {
+    /// PackagedData on the mortar mesh
+    PackagedData packaged_data;
+    /// Unit normal dotted into the fluxes on the face mesh
+    db::item_type<normal_dot_fluxes_tag> normal_dot_fluxes;
```

Comment:
At the moment they are on different meshes, so that wouldn't work.  However, it might be better to change the code to do the calculations with the local n.F on the mortar.  It would make the sending code and the flux-lifting code more complicated, but it would remove this extra storage and simplify the upcoming h-refinement logic.  I will try it and see which seems better.

Reply:
Aren't we storing both on every mortar to do the flux-lifting? Could you explain to me why they are on different meshes?

Reply:
The current calculation being done is roughly `n (P F* - F)` with `F*` on the mortar and `n` and `F` on the face, with `P` the projection operator.  The second method I'm considering is `n P (F* - F)`, which moves `F` to the mortar.  The two expressions are equivalent in the absence of h-refinement.  (The multiplication by the magnitude of the normal must be done on the face mesh.)  The quantities not needed to calculate `F*` are only stored locally, not sent to the neighboring element.

It is easier to collect the data together for the first method, because the mortar-mesh data on both sides is the same (whatever is needed to compute `F*`), and in some cases we can avoid projecting the fluxes to the mortar.  The second method avoids double storage of fluxes in the common case where they are needed for `F*`, and I believe leads to a simpler expression for h-refinement.